docs/user-guides/class_notebooks_best_practices.md

```diff
@@ -1,97 +1,134 @@
+The open-source Python packages used in your Conda environments are routinely updated, often introducing breaking dependencies. If the `environment.yml` file does not pin packages to specific versions, the default behavior is to build the latest available versions. This can introduce new dependency conflicts at any time. 
 
-    1. Distributing ```environment.yml``` file
-    1. Upload ```environment.yml``` into following directory: ```/home/jovyan/conda_environments/Environment_Configs/```         
-    1. Create using ```Create_OSL_Conda_Environments.ipynb``` notebook located in ```/home/jovyan/conda_environments```
+Even if you pin every package, there is still a risk that an `environment.yml` will no longer build correctly at some point. This can occur if one of its package versions is removed due to a security risk, legal issue, or the discovery of a critical error, though this is relatively rare.
 
-- You may encounter dependency conflicts that can prevent you from installing essential software to run all of your notebooks in a single environment. In such cases, create multiple conda environments to run different notebooks.
+### To reduce the chances of your `environment.yml` breaking over time, provide a pinned version to your users:
+1. Open a terminal.
+2. Create the conda environment if it does not yet exist.
+3. Run the command `mamba env export -n your_environement_name > environement.yml`
+4. Open the newly created `environment.yml` in a text editor.
+5. Remove the `prefix` line at the bottom of the file
+    - `prefix` is the path to the environment on your system, which will likely differ between users.
+6. Save the file.
+7. Push the file to your notebook repository.  
 
 ---
 
-## Set the Notebook Metadata to Use the Correct Environment
+## Set The Notebook Metadata to Use the Correct Environment
 
 1. Open your notebook, change into your conda environment's kernel, and save the notebook.
-1. Push the update to your notebook repo.
-1. When students pull in your notebook repo, the notebooks will automatically run the correct kernel with no intervention (as long as the required environment has been created).
+1. Push the update to your notebook repository.
+1. When users clone your repository, the notebooks will automatically use the correct kernel (as long as the required environment has been created).
 
 ---
 
-## Clear Your Notebook Output Before Saving it
+## Consider Clearing Your Notebook Output Before Saving It, Or Not
 
-- Saving a notebook with previous output(s) <!--displayed--> increases its file size and slows down the time it takes to load.
-- [Restart your kernel and clear the notebook output](restarting_server_and_kernel.md) before saving and pushing notebooks to your repo. 
+Saving a notebook with its output increases the file size and slows down the time it takes to open, save, and download. This can cause difficulty for users accessing a JupyterHub over a weak internet connection.
```

Comment:
Since "Or Not" was added, perhaps some context as to why you might _not_ want to clear output might be a useful discussion.

Reply:
Oh, never mind, it on line 41, but the diff sort of masked it.